### ISSUE TITLE
Index out of bounds exception

### ISSUE THREAD

**Issue by wmdietlGC** [typetools/checker-framework#1424]
Take:

```java
abstract class IndexOOB {
  class Box<T> {}

  interface Callable<V> {
    V call() throws Exception;
  }

  class MyCallable<T>
      implements Callable<T> {
    MyCallable(Callable<T> delegate) {}
    public T call() throws Exception {
      return null;
    }
  }

  abstract <T> Box<T> submit(Callable<T> t);

  Box<Boolean> foo() {
    return submit(new MyCallable(() -> true));
  }
}
```

Running the Nullness Checker gives:

```
error: SourceChecker.typeProcess: unexpected Throwable (IndexOutOfBoundsException) while processing IndexOOB.java; message: Index: 0, Size: 0
  Compilation unit: IndexOOB.java
  Exception: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0; Stack trace: com.sun.tools.javac.util.List.get(List.java:476)
  org.checkerframework.framework.type.AnnotatedTypeFactory.makeGroundTargetType(AnnotatedTypeFactory.java:3510)
  org.checkerframework.framework.type.AnnotatedTypeFactory.getFnInterfaceFromTree(AnnotatedTypeFactory.java:3313)
  org.checkerframework.framework.type.AnnotatedTypeFactory.getFnInterfaceFromTree(AnnotatedTypeFactory.java:3299)
  org.checkerframework.common.basetype.BaseTypeVisitor.visitLambdaExpression(BaseTypeVisitor.java:1144)
  org.checkerframework.common.basetype.BaseTypeVisitor.visitLambdaExpression(BaseTypeVisitor.java:160)
  com.sun.tools.javac.tree.JCTree$JCLambda.accept(JCTree.java:1628)
  com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
  org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:262)
  org.checkerframework.common.basetype.BaseTypeVisitor.checkArguments(BaseTypeVisitor.java:2241)
  org.checkerframework.common.basetype.BaseTypeVisitor.visitNewClass(BaseTypeVisitor.java:1113)
  org.checkerframework.checker.nullness.NullnessVisitor.visitNewClass(NullnessVisitor.java:565)
  org.checkerframework.checker.nullness.NullnessVisitor.visitNewClass(NullnessVisitor.java:55)
  com.sun.tools.javac.tree.JCTree$JCNewClass.accept(JCTree.java:1532)
  com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
  org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:262)
  org.checkerframework.common.basetype.BaseTypeVisitor.checkArguments(BaseTypeVisitor.java:2241)
  org.checkerframework.common.basetype.BaseTypeVisitor.visitMethodInvocation(BaseTypeVisitor.java:939)
  org.checkerframework.common.basetype.BaseTypeVisitor.visitMethodInvocation(BaseTypeVisitor.java:160)
  com.sun.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1477)
  com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
  org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:262)
  org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:160)
  com.sun.source.util.TreeScanner.visitReturn(TreeScanner.java:255)
  org.checkerframework.common.basetype.BaseTypeVisitor.visitReturn(BaseTypeVisitor.java:1218)
  org.checkerframework.common.basetype.BaseTypeVisitor.visitReturn(BaseTypeVisitor.java:160)
  com.sun.tools.javac.tree.JCTree$JCReturn.accept(JCTree.java:1390)
  com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
  org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:262)
  org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:160)
```

**Comment by smillst** [comment]
Fixes in #1387.

**Comment by smillst** [comment]
Fixed via a03f6ba.
